packages/zero-solid/src/use-query.ts

```diff
@@ -7,11 +7,21 @@ import type {
   TableSchema,
 } from '../../zero-advanced/src/mod.js';
 import {solidViewFactory} from './solid-view.js';
+import type {ResultType} from '../../zql/src/query/typed-view.js';
+
+export type QueryResultDetails = Readonly<{
+  type: ResultType;
+}>;
+
+export type QueryResult<TReturn extends QueryType> = readonly [
+  Smash<TReturn>,
+  QueryResultDetails,
+];
 
 export function useQuery<
   TSchema extends TableSchema,
   TReturn extends QueryType,
->(querySignal: () => Query<TSchema, TReturn>): Accessor<Smash<TReturn>> {
+>(querySignal: () => Query<TSchema, TReturn>): Accessor<QueryResult<TReturn>> {
```

Comment:
After using this in hello-zero-solid it is clear to me that the return type of this should be:

```ts
[
  Accessor<Smash<TReturn>>,
  Accessor<QueryResultDetails>,
]
```

The way it is done now, one cannot use destructuring which makes the code a bit inconvenient.

Reply:
https://github.com/rocicorp/hello-zero-solid/pull/9